lib/livebook/session.ex

```diff
@@ -1853,6 +1855,14 @@ defmodule Livebook.Session do
     {:noreply, handle_operation(state, operation)}
   end
 
+  def handle_info({:client_disconnected, id}, %{data: %{notebook: %{hub_id: id}}} = state) do
+    # Since the hub got deleted, we need to fallback to Personal hub
+    state = handle_operation(state, {:set_notebook_hub, @client_id, Livebook.Hubs.Personal.id()})
+
+    # and remove secrets from the session
+    {:noreply, %{state | data: %{state.data | secrets: %{}}}}
```

Comment:
All changes to data should happen via operations. Also, we want to only filter out hub secrets, but keep session secrets and startup secrets.

@aleDsz @josevalim do you remember why we convert hub secrets to session secrets, when changing the hub, rather than always just filtering them out?

Reply:
I don’t remember why. Perhaps so it still works if we go back to the previous Hub?

Reply:
That could be the reason, but if you go back and forth and we delete, it only means you need to toggle again, which should be fine, no?

Reply:
I feel that keeping the session and hub secrets sections separate is more consistent, rather than moving secrets form one to the other as the hub changes.

Reply:
@jonatanklosko ideally it would be better to not lose anything once toggling back and forth, especially because people may do that when exploring the deployment options?

Reply:
If they use secrets from a hub, then the selected hub is rather intentional and rather unlikely to change. But since you have a preference, I'm fine with maintaining the current behaviour.

@aleDsz we can have a separate operation like `{:unset_hub_secrets, @client_id}` that filters out all secrets with non-nil `hub_id`.

Reply:
Also, if the current session belongs to a file that is inside a S3 FileSystem, do we crash or do we keep the session open and just report I/O error? If yes, should we close the session?

Reply:
> What if the user "copied" some secrets that has credentials as a session secret?

If you mean manually then we don't need to worry about it, they may as well have copied all secrets to a file :)

> do we crash or do we keep the session open and just report I/O error?

I think this should already be handled, because we resolve the file lazily (on read/write), so if the filesystem can no longer be found, it's going to result in writing error and show up to the user.

Reply:
In the firing user use case, the removed user should no longer have access to hub secrets and hub file systems, but if the user have an open session to a file from org's file system, should we keep this user with access to this file?

I would suggest to drop the session, because the notebook might contain some data (the notebook itself and/or the secrets) which the organization wouldn't like to be accessible after removing the user.

Reply:
I see, I wonder if we should just close the session if it uses the hub that got deleted. This way we don't need to worry about other things like confidential outputs in the current session, and forked sessions. Obviously if the user has a local file, then they can still look at the source, but that's out of our control. @josevalim wdyt?

Reply:
I am OK with closing the session then.